### PR TITLE
Extract definitions to cpp files in shared_model/cryptography

### PR DESCRIPTION
**Pull request by Warchant**
### Description of the Change

- Extract definitions outside of hpp files in `shared_model/cryptography/**/*`. 
- Make cryptography target name `shared_model_cryptography` - neutral to specific algorithm (previous was `shared_model_ed25519_sha3 `)
- Rename `cryptography` target name to `ed25519_crypto` to make it more obvious. `cryptography` is more broad term and covers move than `ed25519_crypto`

### Benefits

This change decreases build time and allows further optimizations -- more hpps can be separated onto hpp/cpp (now we get targets, which we can link to), thus decreasing build time and reducing memory consumption during compilation.

### Possible Drawbacks 

None.

### CODE REVIEW

File: shared_model/cryptography/model_impl/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+    hash.cpp
+    keypair.cpp
+    private_key.cpp
+    public_key.cpp
+    seed.cpp
+    signed.cpp
+    )
+target_link_libraries(shared_model_cryptography_model
+    optional
+    )
```
Comment (Alexey-N-Chernyshov) [inline shared_model/cryptography/model_impl/CMakeLists.txt]: I think here should be an extra line.

Comment (Warchant) [reply shared_model/cryptography/model_impl/CMakeLists.txt]: fixel

File: shared_model/cryptography/blob.hpp [hunk 2]
```diff
@@ -18,20 +18,18 @@
...
 #include "utils/lazy_initializer.hpp"
-#include "utils/string_builder.hpp"
 #include "utils/swig_keyword_hider.hpp"
 
 namespace shared_model {
   namespace crypto {
 
     class Blob;
-    static inline std::string toBinaryString(const Blob &b);
+    inline std::string toBinaryString(const Blob &b);
```
Comment (nickaleks) [inline shared_model/cryptography/blob.hpp]: If we have only a declaration here, why make it inline?

File: shared_model/cryptography/blob.hpp [hunk 3]
```diff
@@ -128,7 +104,7 @@ namespace shared_model {
       std::string hex_;
     };
 
-    static inline std::string   toBinaryString(const Blob &b) {
+    inline std::string toBinaryString(const Blob &b) {
```
Comment (nickaleks) [inline shared_model/cryptography/blob.hpp]: Can we put it inside .cpp?

File: shared_model/cryptography/keypair.hpp [hunk 4]
```diff
@@ -43,49 +45,28 @@ namespace shared_model {
       using PrivateKeyType = PrivateKey;
 
       explicit Keypair(const PublicKeyType &public_key,
```
Comment (nickaleks) [inline shared_model/cryptography/keypair.hpp]: Should it be explicit?

File: shared_model/cryptography/model_impl/blob.cpp [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+#include "cryptography/blob.hpp"
+
+#include "common/byteutils.hpp"
+#include "utils/string_builder.hpp"
+
+namespace shared_model {
+  namespace crypto {
+
+    Blob::Blob(const std::string &blob)
```
Comment (nickaleks) [inline shared_model/cryptography/model_impl/blob.cpp]: Should be explicit (pls check all single argument constructors below)

File: shared_model/cryptography/blob.hpp (hunk 2 above)
Comment (Warchant) [reply shared_model/cryptography/blob.hpp]: https://stackoverflow.com/a/9268589/1953079

Declaration should contain inline.

File: shared_model/cryptography/blob.hpp (hunk 3 above)
Comment (Warchant) [reply shared_model/cryptography/blob.hpp]: No: https://stackoverflow.com/a/9268589/1953079

File: shared_model/cryptography/keypair.hpp (hunk 4 above)
Comment (Warchant) [reply shared_model/cryptography/keypair.hpp]: it was explicit before. I do not change interface or anything, just separate definition and declaration.

**Pull request merged**